src/components/app/App.js

```diff
@@ -54,35 +64,39 @@ class App extends Component {
         <header className='app-title'>
           <h1>Rancid Tomatillos</h1>
         </header>
-        <Route exact path='/' render={() =>{
-            if (this.state.error) {
-              return <Error error={this.state.error} leaveError={this.leaveError}/>
-            } else if (!this.state.movies.length) {
-              return <p>Loading...</p>
-            } else {
-              return <MovieBoard 
-                movies={this.state.movies} 
-                selectMovie={this.selectMovie}
-              />
-            }
-          }}
-        />
-        <Route exact path='/:id' render={({match}) => {
-          const id = parseInt(match.params.id);
-          if (this.state.error) {
-            return <Error error={this.state.error} leaveError={this.leaveError}/>
-          } else if (!this.state.selectedMovie) {
-            return <p>Loading...</p>
-          } else {
-            return <Movie 
-              key={this.state.selectedMovie.id} 
-              movieInfo={this.state.selectedMovie} 
-              selectMovie = {this.selectMovie}
-              unselectMovie={this.unselectMovie}
-              id={id}
-            />
-          }
-        }}/>
+        <Switch>
+          <Route exact path='/' render={() => {
+              return (
+                <>
+                  {this.state.error && <Error error={this.state.error} />}
+                  {!this.state.movies.length && !this.state.error && <p>Loading...</p>}
+                  {!this.state.error && <MovieBoard 
+                    movies={this.state.movies} 
+                    selectMovie={this.selectMovie}
+                  />}
+                </>
+              )
+            }}
+          />
+          <Route exact path='/movies/:id' render={({match}) => {
+            const id = parseInt(match.params.id);
+            return (
+              <>
+                {this.state.error && <Error error={this.state.error} leaveError={this.leaveError}/>}
+                {!this.state.selectedMovie && !this.state.error && <p>Loading...</p>}
+                {!this.state.error && <Movie 
+                  key={this.state.selectedMovie.id} 
+                  movieInfo={this.state.selectedMovie} 
+                  selectMovie = {this.selectMovie}
+                  unselectMovie={this.unselectMovie}
+                  id={id}
+                />}
+              </>
+            )
+            }}
+          />
+          <Route path=''render={() => <Error error='404 Not Found' leaveError={this.leaveError}/>}/>
```

Comment:
Do you think you need that path props? I would be curious if you could omit that